protographic/tests/operations/message-builder.test.ts

```diff
@@ -0,0 +1,386 @@
+        query GetUsers {
+          users {
+            id
+            name
+          }
+        }
+      `);
+
+      const operation = query.definitions[0];
+      if (operation.kind !== 'OperationDefinition' || !operation.selectionSet) {
+        throw new Error('Invalid operation');
+      }
+
+      const usersSelection = operation.selectionSet.selections[0];
+      if (usersSelection.kind !== 'Field' || !usersSelection.selectionSet) {
+        throw new Error('Invalid selection');
+      }
+
+      const typeInfo = new TypeInfo(schema);
+      const userType = schema.getType('User') as GraphQLObjectType;
+
+      const message = buildMessageFromSelectionSet('UsersResponse', usersSelection.selectionSet, userType, typeInfo);
+
+      expect(message.name).toBe('UsersResponse');
+      expect(message.fields.id).toBeDefined();
+      expect(message.fields.name).toBeDefined();
+    });
+
+    test('should handle field aliases', () => {
+      const schema = buildSchema(`
+        type Query {
+          user: User
+        }
+        
+        type User {
+          id: ID!
+          name: String
+        }
+      `);
+
+      const query = parse(`
+        query GetUser {
+          user {
+            userId: id
+            userName: name
+          }
+        }
+      `);
+
+      const operation = query.definitions[0];
+      if (operation.kind !== 'OperationDefinition' || !operation.selectionSet) {
+        throw new Error('Invalid operation');
+      }
+
+      const userSelection = operation.selectionSet.selections[0];
+      if (userSelection.kind !== 'Field' || !userSelection.selectionSet) {
+        throw new Error('Invalid selection');
+      }
+
+      const typeInfo = new TypeInfo(schema);
+      const userType = schema.getType('User') as GraphQLObjectType;
+
+      const message = buildMessageFromSelectionSet('UserResponse', userSelection.selectionSet, userType, typeInfo);
+
+      // Should use actual field names, not aliases
+      expect(message.fields.id).toBeDefined();
+      expect(message.fields.name).toBeDefined();
+    });
+  });
+
+  describe('edge cases', () => {
+    test('should handle empty selection set', () => {
+      const schema = buildSchema(`
+        type Query {
+          ping: String
+        }
+      `);
+
+      const typeInfo = new TypeInfo(schema);
+      const queryType = schema.getQueryType()!;
+
+      const message = buildMessageFromSelectionSet(
+        'EmptyResponse',
+        { kind: Kind.SELECTION_SET, selections: [] },
+        queryType,
+        typeInfo,
+      );
+
+      expect(message.name).toBe('EmptyResponse');
+      expect(message.fieldsArray).toHaveLength(0);
+    });
+
+    test('should skip unknown fields gracefully', () => {
+      const schema = buildSchema(`
+        type Query {
+          user: User
+        }
+        
+        type User {
+          id: ID!
+        }
+      `);
+
+      // This query references a field that doesn't exist
+      const query = parse(`
```

Comment:
Tbh this should return an error. These errors will be caught by a GraphQL parser using its validation rules. We should not swallow them and make it look like it will succeed.